### PR TITLE
feat(split): add query split status interface

### PR DESCRIPTION
**Pull request by hycdong**
This pull request add interface to query app split status.

### CODE REVIEW

File: src/meta/meta_split_service.cpp [hunk 1]
```diff
@@ -293,6 +293,37 @@ void meta_split_service::on_add_child_on_remote_storage_reply(error_code ec,
     parent_context.stage = config_status::not_pending;
 }
 
+void meta_split_service::query_partition_split(query_split_rpc rpc)
+{
+    const std::string &app_name = rpc.request().app_name;
+    auto &response = rpc.response();
+    response.err = ERR_OK;
+
+    zauto_write_lock l(app_lock());
```
Comment (neverchanje) [inline src/meta/meta_split_service.cpp:302]:
```suggestion
    zauto_read_lock l(app_lock());
```

File: src/meta/meta_service.cpp [hunk 2]
```diff
@@ -1031,6 +1034,23 @@ void meta_service::on_control_partition_split(control_split_rpc rpc)
...
+    }
+
+    if (_split_svc == nullptr) {
+        derror_f("meta doesn't support partition split");
+        rpc.response().err = ERR_SERVICE_NOT_ACTIVE;
+        return;
+    }
+    tasking::enqueue(LPC_META_STATE_NORMAL,
+                     tracker(),
+                     [this, rpc]() { _split_svc->query_partition_split(std::move(rpc)); },
```
Comment (neverchanje) [inline src/meta/meta_service.cpp:1050]: Do we have to asynchronously query the table?

File: src/meta/meta_split_service.cpp [hunk 3]
```diff
@@ -293,6 +293,37 @@ void meta_split_service::on_add_child_on_remote_storage_reply(error_code ec,
     parent_context.stage = config_status::not_pending;
 }
 
+void meta_split_service::query_partition_split(query_split_rpc rpc)
```
Comment (neverchanje) [inline src/meta/meta_split_service.cpp:296]:
```suggestion
void meta_split_service::query_partition_split(query_split_rpc rpc) const
```

**Review by neverchanje — APPROVED** [review]

File: src/meta/meta_split_service.cpp [hunk 4]
```diff
@@ -293,6 +293,37 @@ void meta_split_service::on_add_child_on_remote_storage_reply(error_code ec,
...
+    const std::string &app_name = rpc.request().app_name;
+    auto &response = rpc.response();
+    response.err = ERR_OK;
+
+    zauto_read_lock l(app_lock());
+    std::shared_ptr<app_state> app = _state->get_app(app_name);
+    if (app == nullptr || app->status != app_status::AS_AVAILABLE) {
+        response.err = app == nullptr ? ERR_APP_NOT_EXIST : ERR_APP_DROPPED;
+        response.__set_hint_msg(fmt::format(
+            "app({}) {}", app_name, response.err == ERR_APP_NOT_EXIST ? "not existed" : "dropped"));
```
Comment (acelyc111) [inline src/meta/meta_split_service.cpp:307]: Seems there are many code like `app == nullptr ? ERR_APP_NOT_EXIST : ERR_APP_DROPPED` and `response.err == ERR_APP_NOT_EXIST ? "not existed" : "dropped"`, use macros to simplify these code?

**Review by acelyc111 — APPROVED** [review]

**Pull request merged**